src/schemas/Helpdesk/jiraRequests.ts

```diff
@@ -0,0 +1,208 @@
+      logger.error(
+        `could not read Jira Credentials secret at path ${JIRA_ADMIN_VAULT_CREDENTIALS_PATH}`,
+      );
+      throw err;
+    });
+    if (isCredentials(secretData)) {
+      basicAuthString = `${secretData.email}:${secretData.password}`
+    } else {
+      throw new Error(`vault secret at ${JIRA_ADMIN_VAULT_CREDENTIALS_PATH} could not be read`);
+    }
+  } else {
+    basicAuthString = `${JIRA_ADMIN_EMAIL}:${JIRA_ADMIN_PASS}`
+  }
+
+  return `Basic ${Buffer.from(basicAuthString).toString('base64')}`
+};
+
+
+const getJiraRequestHeader = async () => {
+
+  const auth = await getJiraBasicAuth();
+  return {
+    Authorization: auth,
+    Accept: 'application/json',
+    'Content-Type': 'application/json',
+  };
+}
+
+
+export const createServiceRequest = async (customerIdentifier: string, requestTypeId: string, requestText:string, summaryPrependText: string) => {
+
+    const url = `${JIRA_REST_URI}/request`;
+
+    const characterLimit = 60;
+    const summaryText = requestText.length > characterLimit ? `${requestText.substring(0,characterLimit)}...` : requestText;
+
+    // summary is a required field, that cannot accept newlines
+    const request = {
+      raiseOnBehalfOf: customerIdentifier,
+      serviceDeskId: JIRA_SERVICEDESK_ID,
+      requestTypeId: requestTypeId,
+      requestFieldValues: {
+        summary: `${summaryPrependText}: ${summaryText.replace(/(\r\n|\n|\r)/gm, "")}`,
+        description: requestText
+      }
+    }
+    const bodyData = JSON.stringify(request)
+  
+    const response = await fetch(url, {
+      method: 'post',
+      headers: await getJiraRequestHeader(),
+      body: bodyData,
+    })
+
+    .then(restErrorResponseHandler)
+  
+    .then( async response => {
+      return {requestOk: response.ok, statusText: response.statusText}
+
+    });
+
+    return response;
+  };
+
+
+/* -------------------- Alternative Method for Filing Request on Behalf Of a Customer --------------------------- */
+
+
+type OrganizationCustomer = {
+  // relevant details 
+  accountId: string;
+  emailAddress: string;
+  displayName: string;
+}
+type ErrorBody = {
+  i18nErrorMessage: {
+    parameters : Array<string>
+  }
+}
+
+const EMAIL_ALREADY_EXISTS_MESSAGE = 'email : An account already exists for this email'
+
+
+/**
+ * If you ever wish to create a customer using their name, and get back their client ID.
+ * Client IDs are alternative identifiers used for making service requests, opposed to email
+ * This method requires a Jira Customer Organization
+ * @param email 
+ * @param displayName 
+ * @returns The client id of a new or an existing (based on email) customer
+ */
+export const createCustomer = async (email: string, displayName: string) => {
+
+  const url = `${JIRA_REST_URI}/customer`;
+
+  const bodyData = JSON.stringify({
+    displayName: displayName,
+    email: email,
+  });
+
+
+  const response = await fetch(url, {
+    method: 'post',
+    headers: await getJiraRequestHeader(),
```

Comment:
reaching out to vault on every request seems unnecessary, we can just get it once when the app starts up and keep it in memory so there's less reliance on vault during he runtime. 

If we wrap all the functions in this file into a `JiraClient` type object, that is instantiated using a `createJiraClient` function, then `createJiraclient` can get the credential once and this method can use that credential without reaching vault everytime :)